build/Build.md

```diff
@@ -0,0 +1,190 @@
+
+* *Supports passing WSO2 product startup options.*
+(Refer to [this article](https://medium.com/faun/wso2-products-dockerized-remote-debug-dockerized-wso2-products-with-intellij-idea-111ca8ae60c) for more details)
+
+WSO2 has followed an array of [best practices](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/) and [guidelines](https://github.com/docker-library/official-images#review-guidelines) recommended by the Docker community during the implementation of its product Docker resources.
+
+Some of the best practices and guidelines that are followed for Dockerfiles and entry point scripts are explained below. 
+
+* *Minimized container image size:*
+
+    + *Minimizing the number instructions in the dockerfile that would create sizeable image layers,  leading to reduced image size.* <br/> (Refer to [this section](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers) for advanced details)
+
+    + *The product distribution pack and its dependencies are removed to prevent the Docker image size from increasing. <br/> Only the necessary software packages have been installed.* (Refer to [this section](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#dont-install-unnecessary-packages) for more details)
+
+* *Non-root user:* 
+
+    + *WSO2-released Docker images ship with ‘wso2carbon’ non-root user for running WSO2 products.*
+
+* *Heightened Clarity:* 
+
+    + *Due to sorted multi-line arguments.* (Refer to [this section](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments) for more details)
+
+In case of an extension, WSO2 recommends using an official WSO2 product Docker image as the base of that extended Docker image. 
+
+<br/>
+
+## 1.3. Building from source
+
+In case if the preference is to build the images from source, since this gives you greater control over the image and the conditions <br/>affecting its security, WSO2 recommends referring to the official GitHub release of the Docker resources of the relevant <br/>WSO2 product to write your dockerfile and build a WSO2 product Docker image by source.
+
+Also in case if your preferred JDK is different from the default that we pack, you can follow this approach to build your own docker image.
+
+Useful reference points:
+
+[1] [Docker resources for WSO2 API Management](https://github.com/wso2/docker-apim) <br/>
+[2] [Docker resources for WSO2 Integration](https://github.com/wso2/docker-ei) <br/>
+[3] [Docker resources for WSO2 Identity and Access Management](https://github.com/wso2/docker-is)
+
```

Comment:
@DilanUA please refer to the Open Banking Docker [source](https://github.com/wso2/docker-open-banking) as well.

Reply:
Fixed